protocol.md

```diff
@@ -726,6 +727,120 @@ Upload-Length: 11
+can later retrieve the URL for the created resource.
+
+This allows Clients to resume an upload initiated through the [Creation With Upload](#creation-with-upload)
+or [Creation](#creation) extensions for which they did not receive the response with
+the newly created resource's URL (for example due to a broken connection).
+
+If the Server supports this extension, it MUST add `client-tag` to the `Tus-Extension`
+header.
+
+To initiate an upload with Client Tag, Clients MUST include an `Upload-Tag`
+header with the initial `POST` request (as specified in the [Creation](#creation) and
+[Creation With Upload](#creation-with-upload) extensions) made to the Upload URL.
+
+Clients can then send a `HEAD` request with the same `Upload-Tag` header to the
+Upload URL, which - upon success - MUST respond with the `200 OK` or `204 No Content`
+status, and the resource's URL in the response's `Location` header.
+
+A successful response MUST also contain the `Tus-Version` header. An `Upload-Offset` header
+that indicates how many bytes have already been received by the server MAY also be returned.
+
+Clients can then use the returned `Location` to resume the upload.
+
+If an `Upload-Tag` is unknown or no longer known by the Server, it MUST
+respond to `HEAD` requests with the `404 Not Found` or `410 Gone` status.
+
+If an `Upload-Tag` is already in use for a different, ongoing upload, the Server
+MUST respond to the `POST` request initiating the upload with a `409 Conflict` status.
+
+#### Security Considerations
+
+To protect against attackers trying to guess an `Upload-Tag` and subsequently
+use it to inject malicious content into an upload, Servers MUST take measures to
+ensure that only the party that created a resource with an `Upload-Tag` can use it
+for subseqeuent `PATCH` and `HEAD` requests.
+
+If uploading to a Server is only possible for authenticated users, the Server can
+satisfy this requirement by leveraging the available authentication information to
+bind the `Upload-Tag` to a particular user, so that only that user can use it.
+
+Servers MUST return a `403 Forbidden` status for unauthenticated requests that
+contain an `Upload-Tag` header.
+
+#### Headers
+
+##### Upload-Tag
+
+The `Upload-Tag` request header MUST be an identifier created by the
+Client that's unique within its storage space, to avoid collisions.
+
+Clients can satisfy this requirement by generating and using a [version 4 UUID](https://tools.ietf.org/html/rfc4122#section-4.4).
+
+A `Upload-Tag` MUST consist of ASCII characters and MUST NOT exceed 200 bytes.
+
+The `Upload-Tag` header MUST be included with the `POST` request that creates
+the resource. It MUST also be sent with a follow-up `HEAD` request used to retrieve the
+`Location` of the newly created resource.
+
+Servers MUST respond to `PATCH` and `HEAD` requests with a `404 Not Found` status,
+if they include an `Upload-Tag` header identifying an upload resource that was created
+by a different user.
```

Comment:
"They" should probably be replaced with "the client" to not be confused with "servers".

> Servers MUST respond to `PATCH` and `HEAD` requests with a `404 Not Found` status,
> if the client include an `Upload-Tag` header identifying an upload resource that was created
> by a different user.

Reply:
Why is `PATCH` included here? The `Upload-Tag` is used with the `HEAD` request to get the `Location` header which is then used for the `PATCH` request. Am I missing something?

Reply:
There was a suggestion to also use the Upload-Tag for PATCH requests, so that the Client does not have to store the upload URL anymore. Maybe the sentence is a relict from that idea. However, I am not a huge fan of that, so I am in favor of removing PATCH here :)

Reply:
Good catch! I'll remove the `PATCH` requests with the next commit.

I also found that that paragraph (asking for a `404` response) stands in conflict with the last paragraph of the security considerations (asking for a `403` response), so I'm moving the security considerations contents down a bit and unify them with the `Upload-Tag` header spec - with the next commit.